### PR TITLE
feat(tailscale): add status command

### PR DESCRIPTION
**Pull request by erics118**
## Description

adds a command to show the tailscale status
shows the hostname and the connected exit node if there is one

## Screencast

CleanShot 2023-11-09 at 15 22 44

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Thanks for your contribution @erics118 🔥 

@samlinville do you mind checking this?

**Review by samlinville — COMMENTED** [review]
Thanks! Left one request in comments.

File: extensions/tailscale/src/shared.tsx
Comment (erics118) [inline extensions/tailscale/src/shared.tsx:45]: Props were reordered to match the order given by the CLI

File: extensions/tailscale/src/status.tsx
Comment (erics118) [reply extensions/tailscale/src/status.tsx]: I've switched it to show the magic dns suffix, because I didn't find anything better to describe the active tailnet

File: extensions/tailscale/src/devices.tsx
Comment (rosszurowski) [inline extensions/tailscale/src/devices.tsx:15]: Mind reverting this line, since the old version was the correct punctuation? (An apostrophe rather than a hash mark)

File: extensions/tailscale/src/status.tsx
Comment (rosszurowski) [inline extensions/tailscale/src/status.tsx:20]: We throw a few different errors depending on why the status failed:

- `InvalidPathError` is when the `path` preference is set incorrectly
- `NotRunningError` is when Tailscale is installed but open
- `NotConnectedError` is when Tailscale is installed and open, but not connected to any tailnet

Can you update this error message to handle these three errors?

File: extensions/tailscale/src/status.tsx
Comment (rosszurowski) [inline extensions/tailscale/src/status.tsx:15]: I think this can just be `Connected on ${magicDNSSuffix}` because Tailscale is already in the command name.

Screenshot 2023-11-10 at 11 46 46@2x

File: extensions/tailscale/package.json
Comment (rosszurowski) [inline extensions/tailscale/package.json:61]: We can probably also get rid of this "Tailscale" too, since it should already be clear with the icon:

Screenshot 2023-11-10 at 11 49 01@2x

File: extensions/tailscale/package.json
Comment (rosszurowski) [inline extensions/tailscale/package.json:62]:
```suggestion
      "description": "Show Tailscale's connection status.",
```

File: extensions/tailscale/src/exit.tsx
Comment (rosszurowski) [inline extensions/tailscale/src/exit.tsx:87]: This is great, thanks!

**Review by rosszurowski — APPROVED** [review]
Thanks for the change. There's a few small things that would be nice to clean up before merging, but afterwards this is good to go!

File: extensions/tailscale/src/status.tsx
Comment (erics118) [reply extensions/tailscale/src/status.tsx]: I opted for reusing `getErrorDetails` function, so there's less code duplication

**Comment by pernielsentikaer** [comment]
Is this ready for @rosszurowski to have another look @erics118

**Comment by rosszurowski** [comment]
@pernielsentikaer I'll leave it to @erics118 to confirm, but the code LGTM! All the issues got addressed.

**Comment by erics118** [comment]
All good now!

**Review by pernielsentikaer — APPROVED** [review]

**Pull request merged**
